frontend/src/utilities/useWatchIntegrationComponents.tsx

```diff
@@ -0,0 +1,103 @@
+import * as React from 'react';
+import { useAppSelector } from '~/redux/hooks';
+import { OdhApplication } from '~/types';
+import { getIntegrationAppEnablementStatus } from '~/services/integrationAppService';
+import { allSettledPromises } from '~/utilities/allSettledPromises';
+import { POLL_INTERVAL } from './const';
+import { isIntegrationApp } from './utils';
+
+type OdhIntegrationApplication = OdhApplication & { spec: { internalRoute: string } };
+
+export const useWatchIntegrationComponents = (
+  components?: OdhApplication[],
+): { checkedComponents: OdhApplication[]; isIntegrationComponentsChecked: boolean } => {
+  const [isIntegrationComponentsChecked, setIsIntegrationComponentsChecked] = React.useState(false);
+  const forceUpdate = useAppSelector((state) => state.forceComponentsUpdate);
+  const initForce = React.useRef<number>(forceUpdate);
+  const integrationComponents = React.useMemo(
+    () =>
+      components?.filter((component): component is OdhIntegrationApplication =>
+        isIntegrationApp(component),
+      ),
+    [components],
+  );
```

Comment:
After the last two comments about type -- you can revert this back as the check is moved to the utility.
```suggestion
  const integrationComponents = React.useMemo(
    () => components?.filter((component) => isIntegrationApp(component)),
    [components],
  );
```

Reply:
Done